### PR TITLE
deps: update criterion dependency to v0.4.0

### PR DESCRIPTION
**Pull request by koushiro**

### CODE REVIEW

**Comment by pchickey** [comment]
Thanks for this contribution. As part of our supply-chain security vetting, I have performed an audit of these dependency changes in https://github.com/bytecodealliance/wasmtime/pull/5946 (this auditing activity is restricted to core contributors, for security reasons). Once that PR is merged into `main`, you can please rebase on top of it, at which point the `CI / Cargo vet` step should pass.

**Review by alexcrichton — APPROVED** [review]

**Pull request merged**
